src/dbami/helpers/apply_migrations.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import time
+from typing import Literal, Optional, Sequence
+
+import asyncpg
+from buildpg import V, render
+
+from dbami.db import DB
+
+logger = logging.getLogger(__name__)
+
+
+async def grant_connect_privileges(conn: asyncpg.Connection, *role_names: str) -> None:
+    for role_name in role_names:
+        logger.warning("Granting connect privileges for role '%s'", role_name)
+        q, p = render(
+            """
+                DO $_$
+                    BEGIN
+                        EXECUTE FORMAT('GRANT CONNECT on database %s TO :role',
+                        CURRENT_DATABASE());
+                    END
+                $_$;
+            """,
+            role=V(role_name),
+        )
+        await conn.execute(q, *p)
+
+
+async def revoke_connect_privileges(conn: asyncpg.Connection, *role_names: str) -> None:
+    for role_name in role_names:
+        logger.warning("Revoking connect privileges for role '%s'", role_name)
+        q, p = render(
+            """
+                DO $_$
+                    BEGIN
+                        EXECUTE FORMAT('REVOKE CONNECT on database %s FROM :role',
+                        CURRENT_DATABASE());
+                    END
+                $_$;
+            """,
+            role=V(role_name),
+        )
+        await conn.execute(q, *p)
+
+
+async def active_connections_exist(conn: asyncpg.Connection) -> bool:
```

Comment:
won't this always return the current active connection?

Reply:
nm, that's what pg_backend_pid() is. I must have another open connection somewhere.